src/library_wasm_worker.js

```diff
@@ -145,7 +145,9 @@ addToLibrary({
   _emscripten_create_wasm_worker__postset: `
 if (ENVIRONMENT_IS_WASM_WORKER) {
   _wasmWorkers[0] = this;
-  addEventListener("message", _wasmWorkerAppendToQueue);
+  if (typeof AudioWorkletGlobalScope === 'undefined') {
```

Comment:
Can you do `if (!ENVIRONMENT_IS_AUDIO_WORKLET)` instead?

And also wrap this line in `if AUDIO_WORKLET`.    You could either using an if statement without braces to prevent the line being executed, or you could do `if (AUDIO_WORKLET) return;`

Also, perhaps, add a comment here?

Reply:
Done. I'm not sure if you typically squash PRs, so I force-pushed over the single commit in this PR with a new one.